build/build-i18n-module.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license Copyright 2020 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * Rollup plugins don't export types that work with commonjs.
+ * @template T
+ * @param {T} module
+ * @return {T['default']}
+ */
+function rollupPluginTypeCoerce(module) {
+  // @ts-expect-error
+  return module;
+}
+
+const fs = require('fs');
+
+const rollup = require('rollup');
+const {nodeResolve} = require('@rollup/plugin-node-resolve');
+const {terser} = require('rollup-plugin-terser');
+// Only needed b/c getFilenamePrefix loads a commonjs module.
+const commonjs = rollupPluginTypeCoerce(require('@rollup/plugin-commonjs'));
+const replace = rollupPluginTypeCoerce(require('rollup-plugin-replace'));
+const nodePolyfills = rollupPluginTypeCoerce(require('rollup-plugin-polyfill-node'));
+// @ts-expect-error: no types
+const shim = require('rollup-plugin-shim');
+const {LH_ROOT} = require('../root.js');
+
+const distDir = `${LH_ROOT}/dist`;
+const bundleOutFile = `${distDir}/i18n-module.js`;
+const generatorFilename = `./lighthouse-core/lib/i18n/i18n-module.js`;
+
+const localeBasenames = fs.readdirSync(LH_ROOT + '/lighthouse-core/lib/i18n/locales/');
+const actualLocales = localeBasenames
+  .filter(basename => basename.endsWith('.json') && !basename.endsWith('.ctc.json'))
+  .map(locale => locale.replace('.json', ''));
+
+const plugins = [
+  replace({
+    delimiters: ['', ''],
+    values: {
+      '[\'__availableLocales__\']': JSON.stringify(actualLocales),
+      '__dirname': '""',
+    },
+  }),
+  shim({
+    ['./locales.js']: 'export default {}',
```

Comment:
taking out `path` and `lighthouse-logger` will also shrink the i18n module by a ton and they aren't needed for swapping the locale. 82KiB (26KiB after gzip) down to 34KiB (10KiB after gzip)

Reply:
> and they aren't needed for swapping the locale

actually that's not quite true, albeit it shouldn't be possible with a fixed set of locales. If it tries to swap with an unknown locale it will [try to (lighthouse-log](https://github.com/GoogleChrome/lighthouse/blob/577ef75fbab8d259f372c4d10db5b2e2c3511b0a/lighthouse-core/lib/i18n/i18n.js#L171) the problem. Shouldn't be doing that in the browser anyways.

Reply:
I can't work out how to shim it in the i18n chunk and not the main chunk (the viewer app does use logger.. )

Reply:
oh, maybe I didn't test enough, but I just added
```
  'path': 'export default {}',
  'lighthouse-logger': 'export default {}',
```
and the version in `dist/gh-pages/viewer/' loaded and allowed switching locales

Reply:
What I mean is, wouldn't this remove the logging functionality from the rest of the viewer app?